opm/autodiff/AutoDiffBlock.hpp

```diff
@@ -676,6 +677,50 @@ namespace Opm
         return AutoDiffBlock<Scalar>::function( std::move(val), std::move(jac) );
     }
 
+    /**
+     * @brief Computes the value of base raised to the power of exp
+     *
+     * @param base The base AD forward block
+     * @param exp  The exponent AD forward block
+     * @return The value of base raised to the power of exp
+     */    template <typename Scalar>
+    AutoDiffBlock<Scalar> pow(const AutoDiffBlock<Scalar>& base,
+                                    const AutoDiffBlock<Scalar>& exp)
+    {
+        const int num_elem = base.value().size();
+        assert(exp.value().size() == num_elem);
+        typename AutoDiffBlock<Scalar>::V val (num_elem);
+        for (int i = 0; i < num_elem; ++i) {
+            val[i] = std::pow(base.value()[i], exp.value()[i]);
+        }
+
+        // (f^g)' = f^g * ln(f) * g' + g * f^(g-1) * f'
+        typename AutoDiffBlock<Scalar>::V der1 = val;
+        for (int i = 0; i < num_elem; ++i) {
+            der1[i] *= std::log(base.value()[i]);
+        }
+        std::vector< typename AutoDiffBlock<Scalar>::M > jac1 (base.numBlocks());
```

Comment:
This (and parts below) depends on `base.numBlocks() == exp.numBlocks()`. That should be `assert()`ed.

However, in the case when one of the arguments has `derivative().empty() == true` this function will fail even though it is actually a simpler case (one argument is a constant). This should be checked at the start of the function, one can call the already-existing pow() overload if exp is a constant. There seems to be no overload for the other case (constant base raised to variable exponent), that should be added, too.